initial_setup/tui/spokes/eula.py

```diff
@@ -0,0 +1,128 @@
+"""EULA TUI spoke for the Initial Setup"""
+
+import gettext
+import codecs
+import logging
+
+from pyanaconda.ui.tui.spokes import NormalTUISpoke
+from simpleline.render.widgets import TextWidget, CheckboxWidget, SeparatorWidget
+from simpleline.render.containers import ListRowContainer
+from simpleline.render.screen import UIScreen, InputState
+from simpleline.render.screen_handler import ScreenHandler
+from pyanaconda.ui.common import FirstbootOnlySpokeMixIn
+from initial_setup.product import get_license_file_name, eula_available
+from initial_setup.common import LicensingCategory
+from pyanaconda.core.constants import FIRSTBOOT_ENVIRON
+from pykickstart.constants import FIRSTBOOT_RECONFIG
+
+log = logging.getLogger("initial-setup")
+
+_ = lambda x: gettext.ldgettext("initial-setup", x)
+N_ = lambda x: x
+
+__all__ = ["EULAspoke"]
+
+
+class EULAspoke(FirstbootOnlySpokeMixIn, NormalTUISpoke):
+    """The EULA spoke providing ways to read the license and agree/disagree with it."""
+
+    category = LicensingCategory
+
+    def __init__(self, *args, **kwargs):
+        NormalTUISpoke.__init__(self, *args, **kwargs)
+        self.title = _("License information")
+        self._container = None
+
+    def initialize(self):
+        NormalTUISpoke.initialize(self)
+
+    def refresh(self, args=None):
+        NormalTUISpoke.refresh(self, args)
+
+        self._container = ListRowContainer(1)
+
+        log.debug("license found")
+        # make the options aligned to the same column (the checkbox has the
+        # '[ ]' prepended)
+        self._container.add(TextWidget("%s\n" % _("Read the License Agreement")),
+                            self._show_license_screen_callback)
+
+        self._container.add(CheckboxWidget(title=_("I accept the license agreement."),
+                                           completed=self.data.eula.agreed),
+                            self._license_accepted_callback)
+        self.window.add_with_separator(self._container)
+
+    @property
+    def completed(self):
+        # Either there is no EULA available, or user agrees/disagrees with it.
+        return self.data.eula.agreed
+
+    @property
+    def mandatory(self):
+        # This spoke is always mandatory.
+        return True
+
+    @property
+    def status(self):
+        return _("License accepted") if self.data.eula.agreed else _("License not accepted")
+
+    @classmethod
+    def should_run(cls, environment, data):
+        # the EULA spoke should only run in Initial Setup
+        if eula_available() and environment == FIRSTBOOT_ENVIRON:
```

Comment:
Please, call `FirstbootOnlySpokeMixIn.should_run()` to check the environment.

Reply:
Good point, it's even already imported. Changed.